src/Components/Endpoints/src/FormMapping/Factories/FileConverterFactory.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Components.Endpoints.FormMapping;
+
+internal sealed class FileConverterFactory(IHttpContextAccessor httpContextAccessor) : IFormDataConverterFactory
+{
+    [RequiresDynamicCode(FormMappingHelpers.RequiresDynamicCodeMessage)]
+    [RequiresUnreferencedCode(FormMappingHelpers.RequiresUnreferencedCodeMessage)]
+    public bool CanConvert(Type type, FormDataMapperOptions options) => type == typeof(IFormFile) || type == typeof(IFormFileCollection);
+
+    [RequiresDynamicCode(FormMappingHelpers.RequiresDynamicCodeMessage)]
+    [RequiresUnreferencedCode(FormMappingHelpers.RequiresUnreferencedCodeMessage)]
+    public FormDataConverter CreateConverter(Type type, FormDataMapperOptions options)
+    {
+        return Activator.CreateInstance(typeof(FileConverter<>).MakeGenericType(type), httpContextAccessor.HttpContext) as FormDataConverter ??
+            throw new InvalidOperationException($"Unable to create converter for '{type.FullName}'.");
+    }
```

Comment:
You can register the converters directly on the options by default, that removes the need for the factory. (There is a well-known converters class or something like that, where common ones are placed).

Reply:
This is because we have a set of finite T types, as opposed to IParsable, ICollection, etc.

Reply:
> You can register the converters directly on the options by default, that removes the need for the factory. 

It seems like the factory is still needed to evaluate whether or not a type can be converted with a given converter, as in the `FormDataMetadataFactory`. It's totally possible to do a straight type comparison there, but I observed that other converters (even the `IParsable` and `ICollection` ones) call into the factory first.

Another reason I avoided registering the converter in well known converts is to avoid the hassle of passing in the `IHttpContextAccessor` but to your point below we might be able to remove that.